### DISCUSSION TITLE
Does the TextEmbeddingGenerationService use retries on Azure?

### DISCUSSION THREAD

**Discussion by wmeints** [microsoft/semantic-kernel#9188] [Q&A] [answered]
Hi, I'm wondering whether the TextEmbeddingGenerationService for Azure OpenAI is using retries. I keep seeing 429 errors in my logs, but I also see some lines mentioning retries. I checked the code [for the AzureOpenAITextEmbeddingGenerationService](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.AzureOpenAI/Services/AzureOpenAITextEmbeddingGenerationService.cs) for any clues, but it doesn't mention anything around retries. Not even in the base class.

**Comment by sophialagerkranspandey** [comment, marked as answer]
The error you're getting with 429 is due to throttling. You can see the details for limitations within the headers. We recommend adjusting and then retrying after an interval.
